server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -315,6 +315,33 @@ public DirectoryReader getDirectoryReader() {
         return (DirectoryReader) reader;
     }
 
+    /**
+     * Returns the total hit count from index reader leaves and a query using the count function.
+     * The query needs to be already rewritten.
+     * The method returns -1 if the count cannot be computed in sub-linear time.
+     *
+     * @param query the query to compute hit count.
+     * @return the total hit count of a given query.
+     * @throws IOException if there is a low-level I/O error
+     */
+    public int getTotalHitCount(Query query) throws IOException {
+        assert query == rewrite(query);
+        int hitCount = -1;
+        Weight weight = createWeight(query, ScoreMode.COMPLETE_NO_SCORES, 1f);
+        for (LeafReaderContext leaf : this.getIndexReader().leaves()) {
+            int count = weight.count(leaf);
+            if (count != -1) {
+                if(hitCount == -1){
```

Comment:
Let's initialize hitCount to 0 to make things easier to read since we wouldn't need this if statement any longer?